src/expr.rs

```diff
@@ -232,6 +236,7 @@ ast_struct! {
     /// A slice literal expression: `[a, b, c, d]`.
     ///
     /// *This type is available if Syn is built with the `"full"` feature.*
+    #[cfg_attr(syn_enable_doc_cfg, doc(cfg(feature = "full")))]
```

Comment:
These types declared using `#full` can have the attribute applied by the `ast_struct!` macro. https://github.com/dtolnay/syn/blob/696829500ede9aaff0dd74409d9bbae8404defb4/src/macros.rs#L6-L9

I think that every `ast_struct!`, `ast_enum_of_structs!` or `ast_enum!` requires either `feature = "derive"` or `any(feature = "derive", feature = "full")`, so we might be able to get away with specifying these attributes (and perhaps also the old available comments?) in that macro. @dtolnay, would you find that cleaner?

Reply:
I would like to keep it simple and stick with the current approach in the PR for now.

Reply:
OK - nevermind on this comment in that case :-)

Reply:
I thought about sticking the attribute to the expansion of the #full token, but didn't do it in the end. The cause was, that there were some inconsistencies and there is no such token for derive||full.